src/three-components/shader-chunk/normalmap_pars_fragment.glsl.js

```diff
@@ -0,0 +1,44 @@
+export const normalmapChunk = /* glsl */ `
+#ifdef USE_NORMALMAP
+
+	uniform sampler2D normalMap;
+	uniform vec2 normalScale;
+
+	#ifdef OBJECTSPACE_NORMALMAP
+
+		uniform mat3 normalMatrix;
+
+	#else
+
+		// Per-Pixel Tangent Space Normal Mapping
+		// http://hacksoflife.blogspot.ch/2009/11/per-pixel-tangent-space-normal-mapping.html
+
+		vec3 perturbNormal2Arb( vec3 eye_pos, vec3 surf_norm ) {
+
+			// Workaround for Adreno 3XX dFd*( vec3 ) bug. See #9988
```

Comment:
Is this file forked from Three.js? If so, can we document the nature of the changes? (ditto for `lights_fragment_maps.glsl.js`)

Reply:
Yes, I can do that.